### PR TITLE
feat(nodejs): bump nodejs in version.txt

### PR DESCRIPTION
**Pull request by mat-rumian**
PR update nodejs autoinstrumentation in version.txt. 

Issue: https://github.com/open-telemetry/opentelemetry-operator/issues/1682

### CODE REVIEW

**Comment by mat-rumian** [comment]
I'm sorry needed to change version to `0.38.0` as it corresponds to `@opentelemetry/sdk-node`. I'm not sure if it's right. I was expecting this to match package `@opentelemetry/auto-instrumentations-node` which defines versions of the installed instrumentation packages.

File: versions.txt [hunk 1]
```diff
@@ -19,7 +19,7 @@ autoinstrumentation-java=1.24.0
 
 # Represents the current release of NodeJS instrumentation.
 # Should match value in autoinstrumentation/nodejs/package.json
-autoinstrumentation-nodejs=0.37.0
+autoinstrumentation-nodejs=0.38.0
```
Comment (jaronoff97) [inline versions.txt:22]: Do we need to upgrade [this](https://github.com/open-telemetry/opentelemetry-operator/blob/main/autoinstrumentation/nodejs/package.json#L18) as well?

**Review by VineethReddy02 — CHANGES_REQUESTED** [review]
@mat-rumian, I don't think the version should be updated in `versions.txt`. Normally, `versions.txt` is bumped during the release process to publish the latest auto-inst for supported languages. Instead, it should be updated in the `auto-instrumentation` directory.

Here's an example of how the last Node.js bump to the latest version was performed: [https://github.com/open-telemetry/opentelemetry-operator/pull/1626/files](https://github.com/open-telemetry/opentelemetry-operator/pull/1626/files)

Also, [here](https://github.com/open-telemetry/opentelemetry-operator/pull/1640/files) is an example of how python versions are bumped.

**Review by ankit-gautam23 — APPROVED** [review]
LGTM

**Comment by TylerHelmuth** [comment]
This PR is no longer needed since we've upgraded past 0.38.0

**Pull request closed without merging**